### DISCUSSION TITLE
state empty on first render

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2568] [Bug report]
Hi, is it normal the state can be empty on the first render?

I am just calling it in the component like that const profileImage = useTestStore((state) => state.profileImage)

https://github.com/pmndrs/zustand/assets/137432044/eaf4b0e7-b3a7-4ed8-a676-5e470a3b6695

**Comment by dbritto-dev** [comment]
@ImamJanjua context? pure react? nextjs? are you using persist middleware? 

Without that info we can not help you

**Comment by karthyTartLabs** [comment]
@ImamJanjua What is the default state value, it might be an empty value and it sets the value from the local or API state value in the component. Could check it?
